src/lib/djerba/helpers/expression_helper/helper.py

```diff
@@ -0,0 +1,122 @@
+        # Run the R script with output to the workspace
+        cmd = [
+            'Rscript',
+            os.path.join(self.get_module_dir(), 'find_expression.R'),
+            '--enscon ', wrapper.get_my_string(self.ENSCON_KEY),
+            '--genelist', wrapper.get_my_string(self.GENE_LIST_KEY),
+            '--gepfile', gep_abs_path,
+            '--outdir', self.workspace.get_work_dir(),
+            '--tcgadata', wrapper.get_my_string(self.TCGA_DATA_KEY),
+            '--tcgacode', wrapper.get_my_string(self.TCGA_CODE_KEY)
+        ]
+        self.logger.debug("Rscript command: "+" ".join(cmd))
+        #subprocess_runner(self.log_level, self.log_path).run(cmd)
+
+    def preprocess_gep(self, gep_path, gep_reference, tumour_id):
+        """
+        Apply preprocessing to a GEP file; write results to tmp_dir
+        CGI-Tools constructs the GEP file from scratch, but only one column actually varies
+        As a shortcut, we insert the first column into a ready-made file
+        TODO This is a legacy CGI-Tools method, is there a cleaner way to do it?
+        TODO Should GEP_REFERENCE (list of past GEP results) be updated on a regular basis?
+        """
+        # read the gene id and FPKM metric from the GEP file for this report
+        fkpm = {}
+        with open(gep_path) as gep_file:
+            reader = csv.reader(gep_file, delimiter="\t")
+            for row in reader:
+                try:
+                    fkpm[row[self.GENE_ID]] = row[self.FPKM]
+                except IndexError as err:
+                    msg = "Incorrect number of columns in GEP row: '{0}'".format(row)+\
+                          "read from '{0}'".format(gep_path)
+                    self.logger.error(msg)
+                    raise RuntimeError(msg) from err
+        # insert as the second column in the generic GEP file
+        ref_path = gep_reference
+        out_file_name = 'gep.txt'
+        with \
+             gzip.open(ref_path, 'rt', encoding=constants.TEXT_ENCODING) as in_file, \
+             self.workspace.open_file(out_file_name, 'wt') as out_file:
+            # preprocess the GEP file
+            reader = csv.reader(in_file, delimiter="\t")
+            writer = csv.writer(out_file, delimiter="\t")
+            first = True
+            for row in reader:
+                if first:
+                    row.insert(1, tumour_id)
+                    first = False
+                else:
+                    gene_id = row[0]
+                    try:
+                        row.insert(1, fkpm[gene_id])
+                    except KeyError as err:
+                        msg = 'Reference gene ID {0} from {1} '.format(gene_id, ref_path) +\
+                            'not found in gep results path {0}'.format(gep_path)
+                        self.logger.warn(msg)
+                        row.insert(1, '0.0')
+                writer.writerow(row)
+        return self.workspace.abs_path(out_file_name)
```

Comment:
We return the absolute path to our output file, which will be needed for the R script